android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -334,54 +336,34 @@ constructor(
       .filter { it.eventType == EventType.RESOURCE_CLOSURE }
       .forEach { eventWorkFlow ->
         eventWorkFlow.eventResources.forEach { eventResource ->
-          eventResource.planDefinitions?.forEach { planDefinition ->
-            val carePlans =
-              fhirEngine.search<CarePlan> {
-                filter(
-                  CarePlan.INSTANTIATES_CANONICAL,
-                  { value = "${PlanDefinition().fhirType()}/$planDefinition" }
-                )
-                filter(CarePlan.SUBJECT, { value = subject.referenceValue() })
-              }
+          val currentResourceTriggerConditions =
+            eventWorkFlow.triggerConditions.firstOrNull { it.eventResourceId == eventResource.id }
+          val conditionsMet =
```

Comment:
Can we rename it to a more suitable name that informs about the conditions we are referring to here?